deploy/stacks/vpc.py

```diff
@@ -125,6 +126,55 @@ def create_new_vpc(self, cidr, envname, resource_prefix):
             ],
             nat_gateways=1,
         )
+        if restricted_nacl:
+            nacl = ec2.NetworkAcl(
+                self, "RestrictedNACL",
+                vpc=self.vpc,
+                network_acl_name=f'{resource_prefix}-{envname}-restrictedNACL',
+                subnet_selection=ec2.SubnetSelection(subnets=self.vpc.private_subnets + self.vpc.public_subnets),
+            )
+            nacl.add_entry(
+                "entryOutbound",
+                cidr=ec2.AclCidr.any_ipv4(),
+                traffic=ec2.AclTraffic.all_traffic(),
+                rule_number=100,
+                direction=ec2.TrafficDirection.EGRESS,
+                rule_action=ec2.Action.ALLOW
+            )
+            nacl.add_entry(
+                "entryInboundHTTPS",
+                cidr=ec2.AclCidr.any_ipv4(),
+                traffic=ec2.AclTraffic.tcp_port(443),
+                rule_number=100,
+                direction=ec2.TrafficDirection.INGRESS,
+                rule_action=ec2.Action.ALLOW
+            )
+            nacl.add_entry(
+                "entryInboundHTTP",
+                cidr=ec2.AclCidr.any_ipv4(),
+                traffic=ec2.AclTraffic.tcp_port(80),
+                rule_number=101,
+                direction=ec2.TrafficDirection.INGRESS,
+                rule_action=ec2.Action.ALLOW
+            )
+            nacl.add_entry(
+                "entryInboundCustomTCP",
+                cidr=ec2.AclCidr.any_ipv4(),
+                traffic=ec2.AclTraffic.tcp_port_range(start_port=1024, end_port=65535),
+                rule_number=102,
+                direction=ec2.TrafficDirection.INGRESS,
+                rule_action=ec2.Action.ALLOW
+            )
+            nacl.add_entry(
+                "entryInboundAllInVPC",
+                cidr=ec2.AclCidr.ipv4("10.0.0.0/16"),
```

Comment:
How about referring to`self.vpc.cidr` here just in case the user changes the cidr from the default?

Reply:
Good idea! I just added it

Reply:
Tested 👍